docs/features/static_role_permissions.md

```diff
@@ -0,0 +1,130 @@
+# Static Role Permissions
+
+This library provides a highly simplified permission system
+based on a static (hardcoded) definition of roles and permissions.
+
+- This saves performance, because it does not require any database queries to check permissions.
+- This saves complexity, because the permissions are defined in a simple dictionary. No n:m relations, no changes at runtime.
+
+## Installation
+
+1. Add `StaticRolePermissionBackend` to your `AUTHENTICATION_BACKENDS` in the `settings.py`:
+
+```py
+AUTHENTICATION_BACKENDS = [
+    # ... other backends
+    # 'django.contrib.auth.backends.ModelBackend',  <-- replace the default ModelBackend
+    "ambient_toolbox.static_role_permissions.auth_backend.StaticRolePermissionBackend",
+]
+```
+
+2. Set `STATIC_ROLE_PERMISSIONS_PATH` as dotted path, pointing to a dictionary,
+   which defines your permissions per role.
+
+Example:
+
+`settings.py`
+```py
+STATIC_ROLE_PERMISSIONS_PATH = "my_project.conf.permisions.PERMISSIONS_DICT"
+```
+
+`my_project.conf.permisions.py`
+```py
+PERMISSIONS_DICT: dict[str, set[str]] = {
+    "role_1": {"permission_1", "permission_2", ...},
+    "role_2": {...},
+}
+```
+
+In fact, the roles and permissions can be any string, but it is recommended to follow the Django permission
+naming convention (e.g. `polls.view_poll`) to ensure compatibility with the Django ecosystem.
+
+## Simple example
+
+```py
+PERMISSIONS_DICT: dict[str, set[str]] = {
+    "admin": {
+        "auth.view_user",
+        "auth.edit_user",
+    },
+    "customer": {
+        "auth.view_user",
+    },
+}
+
+
+class User(AbstractUser):
+    @property
+    def role(self) -> str:
+        return "admin" if self.is_staff else "customer"
+
+
+User(is_staff=True).has_perm("auth.edit_user")  # --> True (without any db query)
+```
+
+## Example with role field
+
+This library does not require a role field on the user model, but here is an example how you could implement it:
+
+```py
+class UserRoleChoices(models.TextChoices):
+    ADMIN = "ADMIN", "Admin"
+    EMPLOYEE = "EMPLOYEE", "Employee"
+    USER = "USER", "User"
+
+
+class User(AbstractUser):
+    role = models.CharField(
+        max_length=100,
+        choices=UserRoleChoices.choices,
+        default=UserRoleChoices.USER,
+    )
+
+
+PERMISSIONS_DICT: dict[str, set[str]] = {
+    UserRoleChoices.ADMIN: {...},
+    UserRoleChoices.EMPLOYEE: {...},
+    UserRoleChoices.USER: {...},
+}
+```
+
+## System check
+
+When `STATIC_ROLE_PERMISSIONS_ENABLE_SYSTEM_CHECK` is set to `True` (default), this library will register a system check.
+So on start up, Django will go through all permissions in `STATIC_ROLE_PERMISSIONS_PATH` and raise a warning
+if any of the permissions does not exist in the Django permission system.
+This is useful to ensure compatibility with the Django ecosystem and to avoid typos in the permissions.
```

Comment:
question: what happens if I use the toolbox but dont use static permissions? do I get a successful check? might this be an argument in favour of deriving the "enabled" state from having a permission dict?

Reply:
You need to set both `STATIC_ROLE_PERMISSIONS_PATH` + `STATIC_ROLE_PERMISSIONS_ENABLE_SYSTEM_CHECK`

We could also derive it from `STATIC_ROLE_PERMISSIONS_PATH` + add a variable `STATIC_ROLE_PERMISSIONS_DISABLE_SYSTEM_CHECK` to DISABLE the check if needed.

Reply:
> You need to set both STATIC_ROLE_PERMISSIONS_PATH + STATIC_ROLE_PERMISSIONS_ENABLE_SYSTEM_CHECK

Maybe we should add this info in the text? Then I'm happy

Reply:
Think about it this way: People might use the toolbox without wanting to use these permissions. If it would be a separat package, I agree 100%.

Reply:
Agree. It is only activated when `STATIC_ROLE_PERMISSIONS_PATH` is set (as documented). 
But you don't have to set `STATIC_ROLE_PERMISSIONS_ENABLE_SYSTEM_CHECK` in addition to enable the checks. You only need to set `STATIC_ROLE_PERMISSIONS_DISABLE_SYSTEM_CHECK` if you want to disable them.

Reply:
ah great! then we are good. do you think the docs are clear enough on this point? Open question.

Reply:
Apparently not, or your reading skills are not representative :D 

> By default, this library will register a system check if `STATIC_ROLE_PERMISSIONS_PATH` is set.

...

> The system check can be disabled by setting `STATIC_ROLE_PERMISSIONS_DISABLE_SYSTEM_CHECK` to `True` in the settings.

I'd say yes... 🤷